eth/bor_api_backend.go

```diff
@@ -48,35 +49,31 @@ func (b *EthAPIBackend) GetVoteOnRootHash(ctx context.Context, starBlockNr uint6
 		return false, errors.New("Only available in Bor engine")
 	}
 
-	localRootHash, err := api.GetRootHash(starBlockNr, endBlockNr)
-
+	localEndBlock, err := b.BlockByNumber(ctx, rpc.BlockNumber(endBlockNr))
 	if err != nil {
-		return false, err
+		return false, errEndBlock
 	}
 
-	if !b.eth.Miner().Mining() {
-		return localRootHash == rootHash, nil
+	localEndBlockHash := localEndBlock.Hash().String()
+
+	if err != nil {
```

Comment:
do we need this check?

Reply:
No, Forgot to remove it while reframing the code

Reply:
Removed it now.